docs/DeveloperGuide.md

```diff
@@ -1,22 +1,264 @@
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
+- [Apache Commons Lang Library](https://commons.apache.org/proper/commons-lang/) - Used for string utilities.
+- [JUnit 5](https://junit.org/junit5/) - Used for testing.
+- Code adapted from [AddressBook-Level3](https://github.com/se-edu/addressbook-level3).
+
+JavaNinja uses the following tools for development:
+- JUnit - Used for testing.
+- Gradle - Used for build automation.
 
 ## Design & implementation
+  
+### Design 
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+#### Architecture 
+
+Main components of the architecture:
+
+`Main` (consisting of class `JavaNinja`) is in charge of the app launch and shut down. 
+1. At app launch, it initialises the components in the correct order and connects them with each other
+2. At shut down, it shuts down the other components and invokes cleanup methods 
+
+The bulk of the app's work is done by the respective components: 
+1. `Cli`: It handles the user interface of the app 
+2. `Parser`: The command executor 
+3. `QuizManager`: Holds the data of the App in memory
+4. `Storage`: Reads data from, and writes data to, the hard disk.
+
+`Commons` represent a collection of classes used by other components.
+
+How the architecture components interact with each other: 
+
+The Sequence Diagram below shows how the components interact with each other for the scenario where the user issues the command `add Flashcard /q what's 2+3 /a 5`
+
+![image](https://github.com/user-attachments/assets/1dd443de-b538-4fb8-b4d1-3bff4643b3dd)
+
+### QuizManager Class Implementation
+The `QuizManager` class orchestrates the main operations of the quiz application. It manages topics, quizzes, past results, and handles the loading and saving of data. This class serves as the central controller that interacts with other classes like `Topic`, `Quiz`, and `Storage`.
+
+#### **Feature Overview**
+The `QuizManager` class is designed to provide the following key functionalities: 
+
+1. **Manage Topic and Questions**:
+    - Load and save questions by topic, allowing easy organization of quiz content. 
+    - Support for multiple question types, such as multiple-choice, true/false, flashcards, and fill-in-the-blank.
+
+2. **Quiz Execution**:
+    - Enable users to start quizzes on specific topics, with time limits and question limits. 
+    - Track the user’s score and display feedback upon quiz completion.
+
+3. **Persistence of Quiz Data**:
+    - Save quiz questions and past quiz results to files, ensuring data is preserved across sessions.
+    - Load saved quiz data from files when the application starts.
+
+4. **Review Past Results**:
+    - Allow users to review their scores and comments from previous quiz attempts, providing a way to track their progress.
+
+
+#### **Implementation Details**
```

Comment:
Perhaps you could include a class diagram that shows the associations between QuizManager class and other classes?